File: backend/models/report.py
import enum
from .base import Base
from sqlalchemy import ForeignKey, JSON, String, Enum
from sqlalchemy.orm import Mapped, mapped_column


class ReportType(enum.Enum):
    table = "TABLE"
    line = "LINE"
    pie = "PIE"
    bar = "BAR"


class Report(Base):
    name: Mapped[str] = mapped_column(String, nullable=False)
    description: Mapped[str] = mapped_column(String, nullable=True)
    created_by_id: Mapped[int] = mapped_column(ForeignKey("user.id"))
    data_source_id: Mapped[int] = mapped_column(ForeignKey("datasource.id"))
    type: Mapped[ReportType] = mapped_column(Enum(ReportType))
    config: Mapped[dict] = mapped_column(JSON, nullable=False)

File: frontend/types/report.ts
import { GenerateQuery } from '@/services/query'

export enum ReportType {
  table = 'TABLE',
  line = 'LINE',
  pie = 'PIE',
  bar = 'BAR',
  sql = 'SQL',
}

export interface SaveReport {
  name: string
  description?: string | null
  request_schema: object
  report_type?: ReportType
  report_config?: object | null
  data_source_id?: number
}

export interface Report {
  id: number
  created_at: string
  updated_at: string
  name: string
  description?: string | null
  created_by_id: number
  config: GenerateQuery
  data_source_id: number
  type: ReportType
}


